app/controllers/search.js

```diff
@@ -43,15 +43,17 @@ export default Controller.extend({
       searchPhrase: this.inputPhrase,
       clicked: {
         type: 'article', // currently the only displayed type in the search
-        id: result.id,
+        id: result.wikiId + '_' + result.id,
         title: result.title,
         position: result.position + 1, // +1 since we need to start with 1 instead of 0
         thumbnail: false, // we do not show thumbnails on SRP right now
-        wikiId: result.wikiId,
+      },
+      filters: {
+          searchType: this.model.getScope()
       },
       target: 'redirect',
       app: 'mw-mobile',
-      siteId: result.wikiId,
+      siteId: this.wikiVariables.id,
```

Comment:
So we pass the ID of the wiki we're currently on?

Reply:
Yes, and `clicked.id` is a composite key combining wiki id and page id.